crates/completion/src/completions/unqualified_path.rs

```diff
@@ -143,6 +143,14 @@ fn fuzzy_completion(acc: &mut Completions, ctx: &CompletionContext) -> Option<()
         potential_import_name,
         true,
     )
+    .filter(|import_candidate| match import_candidate {
+        Either::Left(ModuleDef::Function(function)) => function.as_assoc_item(ctx.db).is_none(),
+        Either::Left(ModuleDef::Const(const_)) => const_.as_assoc_item(ctx.db).is_none(),
+        Either::Left(ModuleDef::TypeAlias(type_alias)) => {
+            type_alias.as_assoc_item(ctx.db).is_none()
+        }
+        _ => true,
+    })
```

Comment:
Let's push this check down -- from within ImportMap, we can figure this out without extra queries. 

So, lets add an extra param to `find_similar_import`. It's even time to refactor it to CommandObject pattern, a-la FindUsage.

Reply:
> It's even time to refactor it to CommandObject pattern, a-la FindUsage.

IMO if we still plan to move local symbol search into `import_map`, we might just wait for that: with a single place to query things, whole `imports_locator` is not needed, including the method mentioned.

Reply:
That turned out a bit uglier than exlected due to an existential type returned from the method, but done.